src/arithmetic/int_set.cc

```diff
@@ -84,6 +84,25 @@ bool IntSet::can_prove_negative() const {
   return (s_int && is_negative_const(ir::Simplify(s_int->i.max)));
 }
 
+bool IntSet::can_prove_non_positive() const {
+  if (const IntervalSet* s_int = (*this).as<IntervalSet>()) {
+    auto max = ir::Simplify(s_int->i.max);
+    return is_zero(max) || is_negative_const(max);
+  }
+  return false;
+}
+
+bool IntSet::can_prove_non_negative() const {
+  if (const IntervalSet* s_int = (*this).as<IntervalSet>()) {
+    // Any reason why we should or should not use can_prove() to implement
+    // these functions?
```

Comment:
@tqchen

Reply:
I guess `can_prove` may be more general when the boundary contains free variables (something like `abs(n)`). Not sure if it really is. I would vote for using `can_prove` here after checking that it doesn't lead to regressions.

Reply:
The only downside of using `can_prove`  I can think of is increase in compile time. Let's see what @tqchen has to say.